### PR TITLE
`Assessment`: Show ∞ symbol for grading instructions without limit

### PR DESCRIPTION
**Pull request by valeriebucher**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).

#### Client
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client/).
- [x] I added multiple screenshots/screencasts of my UI changes.

### Description

Reducing confusion for displaying '0' as the value for unlimited applicable grading instructions by exchanging it with infinity symbol '∞'.

### Steps for Testing

Prerequisites:
- 1 Editor/Instructor
- 1 Programming/Modeling/Text/File-Upload Exercise with Structured Grading Criteria

1. Create an Exercise 
2. Create grading criteria with `Limit` greater than 0
3. Create grading criteria with `Limit` equal to 0 or undefined `Limit`
4. See that the value for undefined or equal to 0 `Limit` is now an infinity symbol '∞', otherwise the `Limit` is displayed as the defined number

#### Code Review
- [x] Review 1
- [x] Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Screenshots

Updated version:

Bildschirmfoto 2021-12-14 um 12 55 21

Old version:

Components of structured assessment

### CODE REVIEW

**Review by FlorianGlombik — APPROVED** [review]
Tested on ts1, works as described for 0/undefined and defined values such as 2 and should be easier understandable than the current solution.
The code changes look good to me.

<details>
<summary>Screenshot of updated grading criteria</summary>

image
</details>

**Review by cremertim — APPROVED** [review]
tested on ts2, the infinite symbol is displayed as expected. 👍 
tested @lBruchmann @Strohgelaender @Bofan-Zhu @VinnavanSembian

**Review by Bofan-Zhu — DISMISSED** [review]
Looks good. Infinity symbol is displayed for Limit 0 and undefined.

File: src/main/webapp/app/assessment/structured-grading-instructions-assessment-layout/structured-grading-instructions-assessment-layout.component.html [hunk 1]
```diff
@@ -16,11 +16,12 @@
             <table class="table" style="margin: 1px">
                 <tbody>
                     <tr>
-                        <td width="4%" [innerHTML]="setScore(instruction.credits)"></td>
+                        <td [innerHTML]="setScore(instruction.credits)" style="width: 4%"></td>
```
Comment (willergeorg) [inline src/main/webapp/app/assessment/structured-grading-instructions-assessment-layout/structured-grading-instructions-assessment-layout.component.html:19]: Here

File: src/main/webapp/app/assessment/structured-grading-instructions-assessment-layout/structured-grading-instructions-assessment-layout.component.html [hunk 2]
```diff
@@ -16,11 +16,12 @@
...
                         <td [innerHTML]="instruction!.gradingScale! | htmlForMarkdown"></td>
                         <td [innerHTML]="instruction!.instructionDescription! | htmlForMarkdown" [ngbTooltip]="setTooltip(instruction)"></td>
-                        <td width="4%" style="position: relative; white-space: nowrap">
-                            <span>{{ instruction.usageCount }}</span>
+                        <td style="width: 4%; position: relative; white-space: nowrap">
```
Comment (willergeorg) [inline src/main/webapp/app/assessment/structured-grading-instructions-assessment-layout/structured-grading-instructions-assessment-layout.component.html:22]: Here

**Review by VinnavanSembian — DISMISSED** [review]
tested in testing session on ts2. Looks good to me!

**Review by willergeorg — DISMISSED** [review]
Tested on TS1. Worked fine. I have a small suggestion. Whenever possible, try to avoid explicit styling within the html template. I marked two affected spots. Reuse or declare CSS classes for this or use bootstrap classes.

**Review by lBruchmann — APPROVED** [review]
Tested on ts2. I followed the testing steps. Everything works and is displayed as expected. "0" is switched with the "∞" character.

**Review by Strohgelaender — CHANGES_REQUESTED** [review]
Tested on TS2, worked as expected. I agree with the requested changes by @willergeorg.

**Comment by FlorianGlombik** [comment]
@Strohgelaender @willergeorg Are you sure that a class with 4% width exist from bootstrap? I can not find one that would fit in the [documentation](https://getbootstrap.com/docs/5.0/utilities/sizing/)

**Review by Strohgelaender — DISMISSED** [review]
After discussion with @FlorianGlombik I agree that this is one possible exception for the general "style should be separated from html" rule.

**Review by baranimo7 — APPROVED** [review]
Tested on TS1. ∞ is shown correctly.

**Review by MertAksehirlioglu — APPROVED** [review]
Tested on TS2, worked as expected.

**Comment by coaxial1** [comment]
I'd vote for moving the inline styles to scss. We have 3 (!) inline styles in this file.
I've prepared this here real quick if you want to use it: https://github.com/ls1intum/Artemis/commit/e07bd820d0c640271569431762e63805157a1a8f

**Comment by valeriebucher** [comment]
thank you @coaxial1 ! Makes sense and looks good to me. I added your commit.

**Review by ole-ve — APPROVED** [review]
Tested on TS1, works as described. 👍

**Review by willergeorg — APPROVED** [review]
Code changes since last review look good.

**Review by mallgaier — APPROVED** [review]
Tested on ts3. 
The functionality was implemented as described. If no limit is defined, infinity is now displayed instead of 0.

**Review by jpbernius — APPROVED** [review]
:shipit:

**Pull request merged**